src/wp-includes/embed.php

```diff
@@ -359,7 +359,50 @@ function wp_oembed_add_discovery_links() {
  * @since 4.4.0
  */
 function wp_oembed_add_host_js() {
-	wp_enqueue_script( 'wp-embed' );
+	add_filter( 'embed_oembed_html', 'wp_prepend_oembed_host_inline_script_tag' );
+}
+
+/**
+ * Inline oEmbed host script before post embeds.
+ *
+ * The wp-embed script is printed before to ensure that the `message` event listener is added before the iframe'd post
+ * embed is able to be loaded. Additionally, it is inlined to prevent the external script from blocking the page from
+ * continuing to render. The wp-embed script is printed before each post embed and not just before the first instance
+ * because it is possible that the filters may apply on content that is not initially printed.
+ *
+ * @since 5.8.2
+ *
+ * @param string $html Embed markup.
+ * @return string Embed markup with oEmbed host JS prepended.
+ */
+function wp_prepend_oembed_host_inline_script_tag( $html ) {
+	if ( ! preg_match( '/<blockquote\s[^>]*wp-embedded-content/', $html ) ) {
+		return $html;
+	}
+
+	$script  = "<script type='text/javascript'>\n";
+	$script .= "<!--//--><![CDATA[//><!--\n";
+	if ( SCRIPT_DEBUG ) {
+		$script .= file_get_contents( ABSPATH . WPINC . '/js/wp-embed.js' );
+	} else {
+		/*
+		 * If you're looking at a src version of this file, you'll see an "include"
+		 * statement below. This is used by the `npm run build` process to directly
+		 * include a minified version of wp-embed.js, instead of using the
+		 * file_get_contents() method from above.
+		 *
+		 * If you're looking at a build version of this file, you'll see a string of
+		 * minified JavaScript. If you need to debug it, please turn on SCRIPT_DEBUG
+		 * and edit wp-embed.js directly.
+		 */
+		$script .= <<<JS
+		include "js/wp-embed.min.js"
+JS;
+	}
+	$script .= "\n//--><!]]>";
+	$script .= "\n</script>";
```

Comment:
I'm curious what the real benefit is for having `npm run build` do the replacement here (stretching back to the introduction in 43bd3d46c9abeeebe1f09544305cdf5f5a4e513d) and in `_print_emoji_detection_script()`:

https://github.com/WordPress/wordpress-develop/blob/6367c01844e34dd372b999beb0e370f52d8c42d5/src/wp-includes/formatting.php#L5728-L5764`

It seems to me that it would now be better to do something like this: 

```suggestion
	$script = wp_get_inline_script_tag(
		file_get_contents( sprintf( ABSPATH . WPINC . '/js/wp-embed' . wp_scripts_get_suffix() . '.js' ) )
	);
```

Is there really a need for `grunt-includes`? WordPress core is using `file_get_contents()` elsewhere, including in the new `wp_maybe_inline_styles()`. It's much more dev-friendly to not use `grunt-includes`, both because the source code is simpler and it's easier to develop patches.

Reply:
That was a very long time ago. 🙂 IIRC, it was a combination of `wp_get_inline_script_tag()` not existing at the time, and wanting to avoid the additional file read, particularly for sites with opcaching.

I agree that it's much simpler to read and develop using this method, I have no objection to switching.

Reply:
Removed in f2c42a434a2533b3a8687882fc3d42482739d2e3.